stdlib/public/core/Filter.swift

```diff
@@ -199,8 +171,25 @@ public struct LazyFilterCollection<
   ///
   /// - Complexity: O(1).
   public var endIndex: Index {
-    return LazyFilterIndex(
-      _baseElements: _base, base: _base.endIndex, _include: _predicate)
+    return LazyFilterIndex(baseIndex: _base.endIndex)
+  }
+
+  // TODO: swift-3-indexing-model - add docs
+  @warn_unused_result
+  public func next(index: Index) -> Index {
+    return LazyFilterIndex(baseIndex: _filteredNext(index.baseIndex))
+  }
+
+  @inline(__always)
+  private func _filteredNext(baseIndex: Base.Index) -> Base.Index {
```

Comment:
We can't use `private` in the standard library because of a compiler limitation.  Please use `internal`.